### DISCUSSION TITLE
Close all but current buffer

### DISCUSSION THREAD

**Discussion by kolpav** [LunarVim/LunarVim#2725] [Q&A] [answered]
Some time ago there was "Close all but current buffer" shortcut which is no longer present due to change of plugin being used for buffer management. Now I have

```lua
lvim.builtin.which_key.mappings["b"]["e"] = { "<cmd>:%bdelete|edit#|bdelete#<cr>", "Close all but current" }
``` 
which is almost there but its producing errors when you have unsaved buffers but mainly it **resets cursor position**.
Is there a better better way how to do it? **I don't care about unsaved buffers** and I would like them gone after using the shortcut.

**Comment by zalegrala** [comment]
If it is just the cursor that moves and you want to solve for that, I think you could add `mark a|silent!` to the beginning of the command, and then `|norm!` to the end.

**Comment by danielnieto** [comment, marked as answer]
I use the following on my config
```
lvim.keys.normal_mode["<Leader>bo"] = ':%bd!|e #|bd #|normal`"<CR>'
```
It closes all buffers without complaining about saving changes, and leaves your cursor on the same place.
